src/v/cloud_topics/level_one/metastore/metastore.h

```diff
@@ -294,6 +294,29 @@ class metastore {
       const model::topic_id_partition&,
       model::timestamp tombstone_removal_upper_bound_ts)
       = 0;
+
+    struct to_sample_info {
+        model::topic_id_partition tid_p;
+        model::timestamp tombstone_removal_upper_bound_ts;
+    };
+
+    struct compaction_info_response {
+        // The dirty ratio of the log
+        double dirty_ratio;
+        // The earliest dirty timestamp in the log. `std::nullopt` if there is
+        // no such timestamp.
+        std::optional<model::timestamp> earliest_dirty_ts;
+        // Compaction offsets
+        compaction_offsets_response offsets_response;
+    };
+
+    virtual ss::future<std::expected<compaction_info_response, errc>>
+    get_compaction_info(const to_sample_info&) = 0;
+
+    // Vectorized RPC for obtaining compaction state for a number of partitions
+    virtual ss::future<
+      chunked_vector<std::expected<compaction_info_response, errc>>>
```

Comment:
This is sort of an awkward return type. We are assuming indexes line up right? Can we doc that or make this a map of tid_p -> response?

Reply:
Agreed that this feels a bit awkward.. Alternatively add the tidp to the compaction_info_response? And make the return type some `expected<chunked_vector<compaction_info_response>, errc>` or something?

Also, consider hiding the RPCs by omitting failed RPCs from the returned set? Is it important for the caller to know about specific failed RPCs?